Feature/deleting messages

msalo3 opened the pull request.

MarcAtNuRelm committed:
Added regexp for removing messages

MarcAtNuRelm committed:
Added receipt handler field to all messages. Refactored message classes for readability

MarcAtNuRelm committed:
Tests pass for regex matches

MarcAtNuRelm committed:
Added deletion of messages when returning to get objects

raysapida reviewed (approved):
Looks good to me! I like that you made it backward compatible with workflows that don't have a `regex_items` config param or any message that doesn't currently have a `receipt_handle`

raysapida committed:
Update `#delete_message` to not have any arguments passed in

- Added vcr to testing gems
- Create an integration spec for getting messages
- Create an integration spec for getting shipments and deleting messages
in teh same request
